core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala

```diff
@@ -2051,4 +2055,59 @@ class GroupMetadataManagerTest {
     group.transitionTo(CompletingRebalance)
     expectMetrics(groupMetadataManager, 1, 0, 1)
   }
+
+  def addDelay(durationMs: Long)(groupMetadata: GroupMetadata): Unit ={
+    time.sleep(durationMs)
+  }
+
+  @Test
+  def testPartitionLoadMetric(): Unit = {
+    val server = ManagementFactory.getPlatformMBeanServer
+    val mBeanName = "kafka.server:type=group-coordinator-metrics"
+    val reporter = new JmxReporter("kafka.server")
+    metrics.addReporter(reporter)
+
+    assertTrue(server.isRegistered(new ObjectName(mBeanName)))
+    assertEquals(Double.NaN, server.getAttribute(new ObjectName(mBeanName), "partition-load-time-max"))
+    assertEquals(Double.NaN, server.getAttribute(new ObjectName(mBeanName), "partition-load-time-avg"))
+    assertTrue(reporter.containsMbean(mBeanName))
+
+    val groupMetadataTopicPartition = groupTopicPartition
+    val startOffset = 15L
+    val memberId = "98098230493"
+    val committedOffsets = Map(
+      new TopicPartition("foo", 0) -> 23L,
+      new TopicPartition("foo", 1) -> 455L,
+      new TopicPartition("bar", 0) -> 8992L
+    )
+
+    val offsetCommitRecords = createCommittedOffsetRecords(committedOffsets)
+    val groupMetadataRecord = buildStableGroupRecordWithMember(generation = 15,
+      protocolType = "consumer", protocol = "range", memberId)
+    val records = MemoryRecords.withRecords(startOffset, CompressionType.NONE,
+      (offsetCommitRecords ++ Seq(groupMetadataRecord)).toArray: _*)
+
+    def loadWithDelay(duration: Int): Unit = {
+      EasyMock.reset(replicaManager)
+      expectGroupMetadataLoad(groupMetadataTopicPartition, startOffset, records)
+      EasyMock.replay(replicaManager)
+      groupMetadataManager.loadGroupsAndOffsets(groupMetadataTopicPartition, addDelay(duration))
+    }
+
+    // max of one 30sec window
+    val durationMs = List(9000, 3000, 7000, 7000, 7000)
+    durationMs.foreach(loadWithDelay)
+    assertEquals(9000.0, server.getAttribute(new ObjectName(mBeanName), "partition-load-time-max"))
+
+    // last window was complete, so compute new max of this window
+    val durationMs2 = List(6000, 2000, 4000)
```

Comment:
What's the goal here? To make sure each window is calculated independently? We can probably do with a single value here.

Reply:
The goal was to ensure that the metric's sliding window works as expected. However since this should be done in MetricsTest, this could be reduced to a smaller test.